### DISCUSSION TITLE
Database Properties

### DISCUSSION THREAD

**Discussion by aps2011** [pawelsalawa/sqlitestudio#4231] [Ideas]
First of all, congratulations on a first-rate application. It is really helpful having the graphical interface when creating databases & developing queries.
As a bit of background, I am now retired and use SQLite3 and Python for collecting and managing some personal data relating to electricity usage (by half hour), a weather station (5 minute snapshots) and a heat pump (2 minute snapshots). An old RaspberryPi, running in headless mode, is scheduled to pick up the data, validate it, write to the various databases in shared storage.
I reckon I have a reasonable handle on SQL (not so much with DDL) having used it on and off for 30 years. In my time I have written some heavy-duty data extracts and analyses of financial data on large databases to support my accountancy work.

The nub of my question/suggestion is SQLite version control and exposure of the version in the file.
Recently I added SQLiteStudio 3.3.3 to my Windows 10 system having previously used 3.2.1. I wanted to add some views using 'window' functions to get 7 day moving averages but this is not available in SQLite 3.24 (needs 3.28). I kept the older version as the Pi's version of Python 3.38 is limited to SQLite 3.27 and I didn't want to create queries that would fall over on that platform.
Yesterday I wanted to review some of the data and unknowingly opened SQLite 3.2.1 only to find that the more recent databases could not be opened.  This had me worried for a while and it took a few minutes to work out what was wrong.
My suggestion is to have a facility to look at the some of properties of a database taken from the first 100 bytes of the database file. Items reported could include:-

- Header String (SQLITE 3)
- SQLite Version (bytes 96-99)
- Text Encoding (e.g. UTF-8)
- Application Version ID (as per the Pragma command)
- User Version
- Page size (bytes)
- Database size in pages
- File size in bytes (from os file management)

It would also help if SQLiteStudio could say why a database could not be opened!

Again, thanks for developing such a useful tool.

**Comment by pawelsalawa** [comment]
We can read header, but how helpful would that be? I mean the sole fact that the database was modified by more recent version not necessarily means that this is the actual reason you cannot open the database. It can be the reason, but doesn't have to be.

Still, I think your suggestion is good and I definitely want to warn user about possibility, that the problem is with newer version involved in recent modification (if the database header indicates so).

Created #4232 to track it.

Regarding other header properties - most of them can be accessed by `PRAGMA` statements (apart from the file size), but they could be of course presented in some database tooltip with details. It's obviously a nice enhancement. I created #4233 for the same.
